infrastructure/src/main/java/org/corfudb/infrastructure/logreplication/replication/send/LogReplicationSourceManager.java

```diff
@@ -165,7 +137,8 @@ public void stopLogReplication() {
      * Termination of the Log Replication State Machine, to enable replication a JVM restart is required.
      */
     public void shutdown() {
-        if(runtime.isShutdown()) {
+        // Shama check and note which aare the 2 places
+        if(isShutdown) {
```

Comment:
I checked this out...from what I saw, shutdown is called only from RuntimeFsm's Stopped.entry()..which in turn is only when local leadership is lost. In this flow, there is no replication_stop event yet.

While cases for which Replication_Stop is enqueued (apart from shutdown here):
 	1. while enqueuing a force snapshot sync. 
 	2. from the replicating state when remote leader changes or when we exit replicating state because the state transitioned to WAITING_FOR_CONNECTIVITY

for now I am not seeing the circular call here...unless I missed something somewhere